Proposed discussion: Renames project to Cerebra

mikaelbr opened the pull request:
This is a _very important_ PR: Starting the discussion to rename the cerebral debugger to Cerebra! Why? As summerized in the README:

> Cerebral is all about side effect management. A common side effect in JavaScript is mutation. This debugger helps you track these side effects and mutations. Another great tool for tracking mutations is [Charles Xaviers Cerebra located in the heart of Xavier's School For Higher Learning](<http://marvel.wikia.com/wiki/Cerebra_(Mutant_Detector)>). Now you can have your own Cerebra to help those mutants find their way in life.

Or in the new tagline:

> Cerebra is the most powerful mutant detecting system currently available (_for [Cerebral](http://cerebraljs.com)_).

I think its a great fit as the "newest" mutant tracking device in Marvel (current version) is called Cerebra which also is a subset of **Cerebra**l. Read more about cerebra here: http://marvel.wikia.com/wiki/Cerebra_(Mutant_Detector). I don't know how much into the comics we should dive here, because as it is with the multiverses of Marvel thing changes, and I think currently the cerebra isn't used for tracking mutants, but reading mutant memories, but traditionally it was used to enhance telepathic abilities helping Professor X amplify his power to help mutants etc. I think it's fine using the traditional use case for cerebra, for the joke of it.

I don't know what you would want to change and where else to change the name for this to be complete. As of now I've pretty much just done a search and replace. Another thing I'd like to see is a specific version of the Cerebral logo with the "cerebra helmet".

mikaelbr committed:
Renames debugger to cerebra and adds description

mikaelbr commented:
~~FYI: I'm working on solving standard vs prettier.~~ Own PR in time

henri-hulski commented:
I've no opinion about changing the name.
But I think adding prettier should go first in a separate commit/PR. 
Take a look how it is done in Cerebral. There we also use standard and prettier together. Standard for linting and prettier for style. The prettier configuration should also be the same as in Cerebral to have a consistent style.

henri-hulski commented:
And it would be good to have 2 commits for prettier. The first with adding prettier and configuration and the second with the new code style applied to the code.

mikaelbr commented:
Removed the terribleness of the PR to avoid "bikesheding" on styles, shouldn't have been there in the first place. Also made a separate PR for the bugfix (#51). I added prettier in #52

christianalfoni commented:
Is it spelled `Cerebra`? ... not `Cerebro`? I checked the internet and it seems to be `Cerebro`? :)

mikaelbr commented:
Check the description and wiki article from my firsg comment. The previous version was called cerebro, the current updated is called cerebra.

christianalfoni commented:
Is that common knowledge? :) `Cerebra` makes me think of womens underwear for some reason

mikaelbr commented:
If it’s common knowledge or not is hard to say. But it is correct knowledge 🤓 in the movies they still have cerebro, so there’s that, so might be more familiar for the «mainstream».

christianalfoni commented:
Hehe... difficult choice, but I feel very uncomfortable with `Cerebra` due to before mentioned issue ;) I would suggest it is more likely that people who er into x-men would understand the `Cerebro` reference, but people who are not into x-men would not understand `Cerebra`. if that makes sense 😂

mikaelbr commented:
I buy that argument of familiarity – and the point is for people to get the reference so should maximize impact. But the `cerebra` reminding you of a bra and making you uncomfortable, I must be frank, is kind of "childish" 😛 It's kind of like not wanting to name an app Boxer as it reminds you of boxershorts, or insisting on not using the term thunks as it reminds you of trunks. Many people use them, pretty household stuff.

christianalfoni commented:
Haha, yeah, I just made a case for hitting puberty at the age of 34 😂 But yeah, familiarity is my official argument for changing it to `cerebro`

Guria commented:
I just leave it here: https://github.com/KELiON/cerebro

As for me, I am a bit skeptic about renaming. As another non-native I look on `Cerebra` as just `Cerebral` with typo.
Though don't count this message as voice against renaming.

christianalfoni commented:
Maybe we can just make a funny thing out of it? So we keep the formal naming to `cerebral-debugger`, but I can rather ask my friend to make a version of the Cerebral brain with the `cerebro` helmet? So when you fire up the debugger we have a more subtle reference? Can of course use it on the website as well :)

Guria commented:
@christianalfoni I guess it could work 👍

corlaez commented:
Wow, I don't feel like changing the name at all. 

Maybe I am biased since my native language is Spanish and I find it nice that Cerebral writes the same with the same meaning in both languages. 

Also I just like how it sounds (specially in English) and seems better to stick to the name and keep maturing the APIs. Branding has to change, domains etc. I mean I wouldn't change the name unless there is a real good reason and not just realizing there is some cool gadget with a similar name in a comic.

Somehow I think it helps to keep ourselves more serious about the tool, I can't think of a project changing it's name so lightly. That's my opinion

mikaelbr commented:
To be clear: this wasn’t a rfc for changing the cerebral project name, but specifically the debugger helper tool.

christianalfoni commented:
Haha, this was a fun discussion, but yeah... Cerebral is beyond marketing stunts I think 😃
